packages/docusaurus/lib/commands/start.js

```diff
@@ -69,15 +67,18 @@ module.exports = async function start(siteDir, cliOptions = {}) {
     );
   }
 
+  const protocol = process.env.HTTPS === 'true' ? 'https' : 'http';
   const port = await getPort(cliOptions.port);
-  const hotPort = await getPort(port + 1);
   const host = getHost(cliOptions.host);
   const {baseUrl} = props;
+  const urls = prepareUrls(protocol, host, port);
+  const openUrl = normalizeUrl([urls.localUrlForBrowser, baseUrl]);
 
   // Create compiler from generated webpack config.
   let config = createClientConfig(props);
 
   const {siteConfig, plugins = []} = props;
+  config.plugin('hmr').use(HotModuleReplacementPlugin);
```

Comment:
Needed for hot reload

Reply:
Better to add this as a comment into the code directly. I'll add it.